pkg/catalog/catalog.go

```diff
@@ -1185,12 +1185,12 @@ func (c *Catalog) SetGarbageCollectionRules(ctx context.Context, repositoryID st
 	return c.Store.SetGarbageCollectionRules(ctx, graveler.RepositoryID(repositoryID), rules)
 }
 
-func (c *Catalog) PrepareExpiredCommits(ctx context.Context, repository string, previousRunID string) (string, error) {
+func (c *Catalog) PrepareExpiredCommits(ctx context.Context, repository string, previousRunID string) (string, string, string, error) {
```

Comment:
Too many return values - better to use a dedicated struct.
You can create a "GarbageCollectionRunMetadata" struct next to "graveler.GarbageCollectionRules". You can return it already from the garbageCollectionManager and pass it all the way up.

Hopefully it won't create a strange import cycle.